source/ast.rs

```diff
@@ -71,6 +71,20 @@ pub enum Literal {
     String(Vec<u8>)
 }
 
-/// An identifier represents any “entity name”.
+/// A name represents either a variable or an entity name.
 #[derive(Debug, PartialEq)]
-pub struct Identifier<'a>(pub &'a [u8]);
+pub enum Name<'a> {
+    /// A variable.
+    Variable(&'a [u8]),
+    /// An unqualified name, i.e. a name without a namespace, like `Bar`.
+    Unqualified(&'a [u8]),
+    /// A qualified name, i.e. a name in a relative namespace (aliased or not),
+    /// like `Foo\Bar`.
+    Qualified(Vec<&'a [u8]>),
+    /// A relative qualified name, i.e. a name in a relative namespace
+    /// restricted to the current namespace, like `namespace\Foo\Bar`.
+    RelativeQualified(Vec<&'a [u8]>),
```

Comment:
I am not very confident with this name. I wonder if `RestrictedQualified` would not be better.